pilot/pkg/xds/discovery.go

```diff
@@ -649,3 +655,11 @@ func (s *DiscoveryServer) ClientsOf(typeUrl string) []*Connection {
 
 	return pending
 }
+
+func (s *DiscoveryServer) WaitForRequestLimit(ctx context.Context) error {
```

Comment:
Should we make the interval configurable too ( fill_interval - and may rename the other setting tokens_per_fill - for consistency with envoy naming in rate limit ) ?

I was not sure if it should wait 1 sec if bucket is empty - or return immediately so it can try another backend. But for this case I think waiting is better, not sure if the exponential backoff is properly implemented in clients and other intermediaries.